### DISCUSSION TITLE
Passport Facebook Token

### DISCUSSION THREAD

**Discussion by lvothnrv** [strapi/strapi#7977] [Legacy Topics]
Hi friends,
I wanted to propose an idea, it is to integrate passport facebook token in provider. It would help people who want to integrate the facebook connection on a mobile application.
Maybe someone has the solution too I don't know.

**Comment by derrickmehaffy** [comment]
Hi @lvothnrv we do have a small guide on this here: https://strapi.io/documentation/v3.x/guides/jwt-validation.html

Naturally due to the almost unlimited possibilities of this we can't give an exact step by step, but instead are only able to point you in the right direction. I hope this helps!

**Reply by lvothnrv** [reply]
Okay I understand there is no problem. I'll see and if I can get it up and running I will share it. Thank you.

**Comment by lvothnrv** [comment]
I found how to connect me via my token coming from a mobile application directly on the facebook provider already included in Strapi. I took a look at how Purest worked and the code of Providers.js (users-permissions / services / Providers.js) and saw that it took either the access_token or the code or oauth_token as a parameter. When we use the Facebook provider via a site it does everything automatically and brings us the user token etc. When we go through a mobile application thanks to the facebook sdk, we can have the access_token and I was looking for how to use with Strapi. After some research I arrive at what I wanted to do and all that without problem thanks to Strapi himself.
For that you just have to use the link: http://localhost:1337/auth/facebook/callback?access_token={yourToken}

Thank you Strap i! Really amazing this Headless CMS !

**Reply by lauriejim** [reply]
Thank you for this comment, I really appreciate it 🚀
